searx/search/processors/online.py

```diff
@@ -0,0 +1,171 @@
+import threading
+import requests.exceptions
+from time import time
+
+from searx.engines import settings
+from searx import poolrequests
+from searx import logger
+from searx.utils import gen_useragent
+
+from searx.search.processors.abstract import EngineProcessor
+
+
+logger = logger.getChild('search.processor.online')
+
+DEFAULT_PARAMS = {
+    'method': 'GET',
+    'headers': {},
+    'data': {},
+    'url': '',
+    'cookies': {},
+    'verify': True
+}
+
+
+class OnlineProcessor(EngineProcessor):
+
+    engine_type = 'online'
+
+    def __init__(self, engine, engine_name):
+        super().__init__(engine, engine_name)
+        self.continuous_errors = 0
+        self.suspend_end_time = 0
+
+    def get_params(self, search_query, engineref):
+        params = super().get_params(search_query, engineref)
+        if params is None:
+            return None
+
+        # skip suspended engines
+        if self.suspend_end_time >= time():
+            logger.debug('Engine currently suspended: %s', self.engine_name)
+            return None
+
+        # add default params
+        params.update(DEFAULT_PARAMS)
+
+        # add an user agent
+        params['headers']['User-Agent'] = gen_useragent()
+
+        # add params from search_query (or engine)
+        if hasattr(self.engine, 'language') and self.engine.language:
+            params['language'] = self.engine.language
+        else:
+            params['language'] = search_query.lang
+
+        params['safesearch'] = search_query.safesearch
+        params['time_range'] = search_query.time_range
+        return params
```

Comment:
```language```, ```safesearch``` and ```time_range``` could be set in the ```abstract.py```, so ```params``` would have these values even for the offline engines.

Reply:
Done with code pushed force (based on the master branch).